src/common/components/SaveButton/SaveButton.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import {TouchableOpacity, Text} from 'react-native';
+import {connect} from 'react-redux';
+
+import {saveCoolers} from '../../../screens/Rooms/actions/roomsActions';
+
+import styles from './saveButtonStyles';
+
+const SaveButton = connect()(({dispatch, navigation}: any) => {
+	const {coolers, roomID} = navigation.state.params;
+
+	if (!coolers) return null;
+
+	const onPress = () => {
+		dispatch(saveCoolers(coolers, roomID));
+		navigation.navigate('CoolersAtRoom', {roomID: navigation.getParam('roomID')})
+	}
+
+	return (
+		<TouchableOpacity
+			onPress={() => onPress()}
```

Comment:
it's a bas solution.on each render you create a new function. move it to body of your component